random-beans/src/main/java/io/github/benas/randombeans/util/FastClasspathScannerFacade.java

```diff
@@ -68,8 +68,9 @@
     }
 
     private static <T> List<Class<?>> searchForPublicConcreteSubTypesOf(final Class<T> type) {
-        List<String> subTypes = type.isInterface() ? scanResult.getNamesOfClassesImplementing(type) : scanResult.getNamesOfSubclassesOf(type);
-        return subTypes.stream().map(className -> {
+        String typeName = type.getCanonicalName();
+        ClassInfoList subTypes = type.isInterface() ? scanResult.getClassesImplementing(typeName) : scanResult.getSubclasses(typeName);
+        return subTypes.getNames().stream().map(className -> {
             try {
                 ClassLoader classloader = Thread.currentThread().getContextClassLoader();
```

Comment:
You really need to not do your own classloading here, since in some situations (e.g. when scanning Spring-Boot jars on the classpath, when the scanner is not itself running within the Spring-Boot classloader), FastClasspathScanner has to create its own `URLClassLoader` to load the class. The correct way to load classes is by calling `ClassInfo#loadClass()`, or you can do it for the whole list by calling `ClassInfoList#loadClasses()` to get a `List<Class<?>>`.

Reply:
Also, the filter line below (`.filter(Objects::nonNull).filter(currentSubType -> isPublic(currentSubType) && !(isAbstract(currentSubType)))`) should be turned into a `ClassInfoFilter` operator that filters the classes *before* classloading, not after, so that only the bare minimum number of classes is actually loaded.